test/routes/germlineSmallMutation/users.test.js

```diff
@@ -0,0 +1,249 @@
+const HTTP_STATUS = require('http-status-codes');
+const supertest = require('supertest');
+const getPort = require('get-port');
+const uuidv4 = require('uuid/v4');
+
+const db = require('../../../app/models');
+const CONFIG = require('../../../app/config');
+const {listen} = require('../../../app');
+
+CONFIG.set('env', 'test');
+const {username, password} = CONFIG.get('testing');
+
+let server;
+let request;
+
+
+const checkGermlineUser = (germlineUserObject) => {
+  expect(germlineUserObject).toEqual(expect.objectContaining({
+    ident: expect.any(String),
+    createdAt: expect.any(String),
+    updatedAt: expect.any(String),
+    role: expect.any(String),
+    user: expect.objectContaining({
+      ident: expect.any(String),
+      createdAt: expect.any(String),
+      updatedAt: expect.any(String),
+      username: expect.any(String),
+      type: expect.any(String),
+      firstName: expect.any(String),
+      lastName: expect.any(String),
+      email: expect.any(String),
+    }),
+  }));
+  expect(germlineUserObject).toEqual(expect.not.objectContaining({
+    id: expect.any(Number),
+    germlineReportId: expect.any(Number),
+    user_id: expect.any(Number),
+    addedById: expect.any(Number),
+    updatedBy: expect.any(Number),
+    deletedAt: expect.any(String),
+  }));
+};
+
+const checkGermlineUsers = (germlineUsers) => {
+  germlineUsers.forEach((germlineUser) => {
+    checkGermlineUser(germlineUser);
+  });
+};
+
+// Start API
+beforeAll(async () => {
+  const port = await getPort({port: CONFIG.get('web:port')});
+  server = await listen(port);
+  request = supertest(server);
+});
+
+describe('/germline-small-mutation-reports/{gsm_report}/users', () => {
+  let user;
+  let report;
+  let createUser;
+  let germlineUserBinding;
+
+  beforeAll(async () => {
+    // Get test user
+    user = await db.models.user.findOne({where: {username}});
+
+    // create a germline report to be used in tests
+    report = await db.models.germlineSmallMutation.create({
+      patientId: 'PATIENT1234',
+      biopsyName: 'TEST BIOPSY NAME',
+      sourceVersion: 'TEST VERSION',
+      sourcePath: 'TEST SOURCE PATH',
+      biofxAssignedId: user.id,
+    });
+
+    createUser = await db.models.user.create({
+      username: uuidv4(),
+      type: 'bcgsc',
+      firstName: 'germlineUserFirstName',
+      lastName: 'germlineUserLastName',
+      email: 'fake@email.com',
+    });
+
+    germlineUserBinding = await db.models.germlineReportUser.create({
+      user_id: createUser.id,
+      germlineReportId: report.id,
+      role: 'clinician',
```

Comment:
This won't actually affect anything, but this should probably be a role that's not clinician or collaborator since they don't have access to germline reports. Projects maybe?

Reply:
Just to make this change as straight forward as possible, I just copied how it is currently being done for the genomic reports. I'm pretty sure these roles are currently not being used for anything though.

However, I can not swap the role to projects because that is not a valid report role based on the role enum in the model
app/models/germlineSmallMutation/users.js

I guess I could add projects to the enum. I'm just hesitant because like I said these roles really aren't used for anything, they were just copied over for simplicity.

@elewis2 Thoughts?

Reply:
What do you mean when you say it won't affect anything? Could it not just be removed in that case? I'm presumably missing something

Reply:
> What do you mean when you say it won't affect anything? Could it not just be removed in that case?
I haven't looked into it too much, but I'm pretty sure it is not currently being used.
This ticket/pull request is to make germline reports like genomic reports, so that the same acl code will work for both of the reports. Since, this is a fairly large change already and removing that section for genomic reports in not in the scope of this ticket. I decided to just maintain how it is done for now with both reports. If there is no plan to use the role property of the binding in the near future, it can most likely be removed from both germline and genomic reports in a separate ticket/pull request.